twenty-first/src/shared_math/tip5.rs

```diff
@@ -909,6 +912,37 @@ mod tip5_tests {
         );
     }
 
+    #[test]
+    fn hash_var_len_equivalence_corner_cases() {
+        // Verify that `hash_var_len` is equivalent
+        // to pad + repeated absorb + squeeze once
+        for preimage_length in [0, 1, 2, 3, 4, 5, 6, 9, 10, 11] {
+            let preimage = random_elements(preimage_length);
```

Comment:
Let's make this a `#[proptest]`, too. In my opinion, we should never use system randomness directly in tests. Randomness should alway be controlled by the testing framework.